### PR TITLE
v4.1.x: Use newer config.guess / config.sub files when relevant

### PR DESCRIPTION
**Pull request by jsquyres**
Refs #8410.

This is the v4.1.x equivalent of master PR #8417.

I chose to leave the reference to the "3rd-party" tree in the code, even though that directory does not exist in the v4.1.x branch.  It just makes the cherry-pick cleaner.

See individual commit messages for more details.

### CODE REVIEW

File: autogen.pl [hunk 1]
```diff
@@ -1137,6 +1137,102 @@ sub in_tarball {
...
+    my @files;
+    File::Find::find(sub {
+        push(@files, $File::Find::name)
+            if ($_ eq "config.guess" ||
+                $_ eq "config.sub") }, $topdir);
+
+    foreach my $file (@files) {
+        # Skip anything in the 3rd-party tree
+        next
+            if ($file =~ /\/3rd-party\//);
```
Comment (bwbarrett) [inline autogen.pl:1225]: technically, this is useless on the 4.x branches, but not worth the one-off.

**Review by bwbarrett — APPROVED** [review]

File: autogen.pl (hunk 1 above)
Comment (jsquyres) [reply autogen.pl:1225]: Yeah, I mentioned that in the PR description.  👍

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ad4b4a5782e6740bdf3ba0f71dc7f22c

**Comment by jsquyres** [comment]
bot:ibm:retest

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/37be967a7141b92189503be0e2c704cb

**Comment by jsquyres** [comment]
@jjhursey The IBM CI has failed twice, and it kinda looks like a "filesystem full" type of error.  But we're only seeing the error in `make install`, not `make all`.  Can you have a look?

**Comment by jjhursey** [comment]
I'm looking. It's an odd failure mode, almost like it was getting files deleted out from under it. The /tmp has hundreds of GB of space free so that shouldn't have been the problem.

**Comment by jjhursey** [comment]
bot:ibm:xl:retest

**Comment by jjhursey** [comment]
The machine that hosts IBM Ci had a file system (and then network) blip this afternoon, as best as I can tell. It builds OMPI in /tmp (local disk) but installs on a parallel file system (for the scaling run tests). The parallel file system blinked out for a moment in a not great way. So that would explain why 'make all' worked but 'make install' threw the error.

Ci is still running slower than normal, but running. Once the system quiets down I'll re-mount some things to see if that brings us back to normal performance.

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/df17845bc3cfcf18fb05a61d0cd603e5

**Comment by jjhursey** [comment]
Let's try one more time 🤞 
bot:ibm:xl:retest

**Pull request merged**